cli/src/lib/stubUtils.js

```diff
@@ -27,16 +27,52 @@ export function glob(pattern: string, options: Object): Promise<Array<string>> {
   );
 }
 
-function resolve(name: string, options: Object): Promise<string> {
-  return new Promise((resolve, reject) =>
-    resolveAsync(name, options, (err, result) => {
-      if (err) {
-        reject(err);
+async function resolveToDirPath(
+  name: string,
+  options: Object
+): Promise<string> {
+  let pkgFiles = [];
+  let opts = {
+    ...options,
+    packageFilter: function(pkgJson, pkgFilePath) {
+      if (!pkgJson.main) {
+        pkgFiles.push(path.join(path.dirname(pkgFilePath), 'index.js'));
+        pkgJson.main = "index.js";
+      }
+      return pkgJson;
+    },
+    isFile: function(filePath, cb) {
```

Comment:
Nit: Given that the return type of `isFile` doesn't matter, could you write it as an async function? Then you could just do

```JavaScript
try {
  const stat = await fs.stat(filePath);
  cb(null, stat.isFile() || stat.isFIFO());
} catch (e) {
  if (e.code === 'ENOENT') {
    cb(null, false);
  } else {
    cb(e);
  }
}
```

The less I see of the Promise API the happier I am :)